hicommonwealth/commonwealth
# User-Onboarding: Auth Type guidance modal

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6642

## Description of Changes
Added `AuthTypeGuidanceModal` that shows up when the user is trying to sign in with an address, that doesn't yet exist on Common.



https://github.com/hicommonwealth/commonwealth/assets/51641047/ee634bfa-19f9-4201-8600-37ff18a29b04



## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_USER_ONBOARDING_ENABLED`
- Logout and select a wallet address (from the wallet extension) that is already connected to common
- Click on the `Sign-in` (important) button, and verify you are able to sign in correctly
- Logout and select a wallet address (from the wallet extension) that is not yet connected to common
- Click on the `Sign-in` (important) button, and verify that when you click on the wallet option (ex: `Metamask`), you get redirected to this modal, this is the `AuthTypeGuidanceModal`
![image](https://github.com/hicommonwealth/commonwealth/assets/51641047/1f9af792-eab0-4136-955d-a296cf9212ec)
- Verify that when you click on `Sign in another way`, you get redirected back to the sign-in modal. Repeating the sign-in process with the same not-connected-before address should lead you back to this modal.
- Now when you are back to the `AuthTypeGuidanceModal`, verify that when you click on `Create an account`, you get redirected to create account modal. When there, you can create an account like normal.
- Logout and select another wallet address (from the wallet extension) that is not yet connected to Common
- Now this time, click on the `Create account` button and verify that you can create an account like normal. The `AuthTypeGuidanceModal` won't show up when creating an account with an address that doesn't yet exist on Common.
- Logout, and click on the `Sign in` button and try to sign in with any SSO option that you haven't connected to Common before
- Verify you get redirected to the `AuthTypeGuidanceModal` modal
- Verify that when you click on `Sign in another way`, you get redirected back to the sign-in modal. Repeating the sign-in process with the same not-connected-before address should lead you back to this modal.
- Now when you are back to the `AuthTypeGuidanceModal`, verify that when you click on `Create an account`, you get redirected to create account modal. When there, you can create an account like normal.
- Mix and match different page states, wallets etc, and verify there is no strange behaviour


## Deployment Plan
N/A

## Other Considerations
Some SSO options seem to be broken on `master` branch. This PR only deals with checking if a new SSO is being used.
